test/images/agnhost/connect/connect.go

```diff
@@ -125,3 +128,44 @@ func connectSCTP(dest string, timeout time.Duration) {
 		os.Exit(1)
 	}
 }
+
+func connectUDP(dest string, timeout time.Duration) {
+	if _, err := net.ResolveUDPAddr("udp", dest); err != nil {
+		fmt.Fprintf(os.Stderr, "DNS: %v\n", err)
+		os.Exit(1)
+	}
+
+	conn, err := net.Dial("udp", dest)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "OTHER: %v\n", err)
+		os.Exit(1)
+	}
+
+	if timeout > 0 {
+		if err = conn.SetDeadline(time.Now().Add(timeout)); err != nil {
+			fmt.Fprintf(os.Stderr, "OTHER: %v\n", err)
+			os.Exit(1)
+		}
+	}
+
+	if _, err = conn.Write([]byte("hostname\n")); err != nil {
```

Comment:
we can't assume the other end knows what `hostname` means, 
this is the problem with UDP, it is connection less, so we need to send something and the other reply in order to know the connection is ok.
What if we add another flag to be able to send data  and we default it to `hostname` or `hello`
```
		cmd = []string{"/agnhost", "connect", fmt.Sprintf("%s:%d", addrTo, toPort), "--timeout=1s", "--protocol=udp","--data=hostname"}
```

Reply:
sure, what about `--udp-data`, since it's only used for UDP?

Reply:
yeah, sounds much better, but keep it defaulted with some data